be/src/exprs/vectorized/cast_expr_array.cpp

```diff
@@ -135,27 +246,39 @@ StatusOr<ColumnPtr> CastStringToArray::evaluate_checked(ExprContext* context, ve
 
     // 3. Assemble elements into array column
     ColumnPtr res = ArrayColumn::create(elements, offsets);
-    if (column->is_nullable()) {
+
+    if (column->is_nullable() || has_null) {
         res = NullableColumn::create(res, null_column);
     }
 
     return res;
 }
 
-Slice CastStringToArray::_unquote(Slice slice) {
+template <bool throw_exception_if_err>
+Slice CastStringToArray<throw_exception_if_err>::_trim(Slice slice) const {
     while (slice.starts_with(" ")) {
         slice.remove_prefix(1);
     }
     while (slice.ends_with(" ")) {
         slice.remove_suffix(1);
     }
+    return slice;
+}
+
+template <bool throw_exception_if_err>
```

Comment:
In this way, all functions of `Expr` need to carry a template parameter, which is not elegant.